docs/tutorials/authentication-github.rst

```diff
@@ -0,0 +1,373 @@
+.. _tutorial-first-steps:
+
+How to setup Github authentication ?
+####################################
+
+In this tutorial, we will authenticate users using Github.
+
+Users obtain a Bearer token from Github and use it in an ``Authenticatìon`` header.
+Leveraging Kinto pluggability, a custom authentication policy is specified in settings in order to validate it.
+
+Custom authentication
+=====================
+
+Create a file :file:`kinto_github.py` with the following scaffold:
+
+.. code-block:: python
+
+    from pyramid.authentication import CallbackAuthenticationPolicy
+    from pyramid.interfaces import IAuthenticationPolicy
+    from zope.interface import implementer
+
+    @implementer(IAuthenticationPolicy)
+    class GithubAuthenticationPolicy(CallbackAuthenticationPolicy):
+        def __init__(self, realm='Realm'):
+            self.realm = realm
+
+        def unauthenticated_userid(self, request):
+            user_id = self._get_credentials(request)
+            return user_id
+
+        def forget(self, request):
+            return [('WWW-Authenticate', 'Bearer realm="%s"' % self.realm)]
+
+        def _get_credentials(self, request):
+            authorization = request.headers.get('Authorization', '')
+            print('Check Github')
+
+
+Don't be scared by those lines. It just implements the necessary methods to match the `IAuthenticationPolicy <http://docs.pylonsproject.org/projects/pyramid/en/latest/api/interfaces.html#pyramid.interfaces.IAuthenticationPolicy>`_ Pyramid interface.
+
+
+Add it to Python path
```

Comment:
You can use the ``add_path_to_virtualenv`` command au wait it just works with ``pew`` - ``pew add /path/to/folder`` nevermind.